Popper type definitions errors

## Bug report

### Current Behavior

When I run `tsc --no-emit` in my project using `@radix-ui/react-tooltip`, I get the following errors:
```
node_modules/@radix-ui/react-popper/dist/index.d.ts:3:10 - error TS2440: Import declaration conflicts with local declaration of 'Arrow'.

3 import { Arrow } from "@radix-ui/react-arrow";
           ~~~~~

node_modules/@radix-ui/react-popper/dist/index.d.ts:17:6 - error TS2456: Type alias 'PopperArrowOwnProps' circularly references itself.

17 type PopperArrowOwnProps = Merge<Polymorphic.OwnProps<typeof Arrow>, {
        ~~~~~~~~~~~~~~~~~~~

node_modules/@radix-ui/react-popper/dist/index.d.ts:20:29 - error TS4109: Type arguments for 'ForwardRefComponent' circularly reference themselves.

20 type PopperArrowPrimitive = Polymorphic.ForwardRefComponent<Polymorphic.IntrinsicElement<typeof Arrow>, PopperArrowOwnProps>;
                               ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
```

### Expected behavior

No such errors to occur.

### Reproducible example

You can run `yarn validate:types` in the `feat-tooltip-component` branch of this repository: `https://github.com/Atom-Learning/components`

### Suggested solution

This error is resolved by replacing `import { Arrow } from "@radix-ui/react-arrow";` at line `3`of `index.d.ts` with `import { Arrow as AnythingElse } from "@radix-ui/react-arrow";` and updating references to the import on lines `17` and `19`.

### Your environment

| Software         | Name(s) | Version |
| ---------------- | ------- | ------- |
| Radix Package(s) |    `@radix-ui/react-tooltip`     |  0.0.7       |
| React            | n/a     |   17.0.1      |
| Browser          |     n/a    | n/a        |
| Assistive tech   |     n/a    |   n/a      |
| Node             | n/a     |     v14.4.0    |
| npm/yarn         |    yarn     |     1.22.5    |
| Operating System |     MacOS Big Sur    |   11.1      |

--- a/.yarn/versions/75172f8d.yml
+++ b/.yarn/versions/75172f8d.yml
@@ -1,0 +1,14 @@
+releases:
+  "@radix-ui/react-accessible-icon": patch
+  "@radix-ui/react-accordion": patch
+  "@radix-ui/react-alert-dialog": patch
+  "@radix-ui/react-context-menu": patch
+  "@radix-ui/react-dropdown-menu": patch
+  "@radix-ui/react-menu": patch
+  "@radix-ui/react-popover": patch
+  "@radix-ui/react-popper": patch
+  "@radix-ui/react-toolbar": patch
+  "@radix-ui/react-tooltip": patch
+
+declined:
+  - primitives

--- a/packages/react/accessible-icon/src/AccessibleIcon.tsx
+++ b/packages/react/accessible-icon/src/AccessibleIcon.tsx
@@ -1,5 +1,5 @@
 import * as React from 'react';
-import { VisuallyHidden } from '@radix-ui/react-visually-hidden';
+import * as VisuallyHiddenPrimitive from '@radix-ui/react-visually-hidden';
 
 const NAME = 'AccessibleIcon';
 
@@ -20,7 +20,7 @@ const AccessibleIcon: React.FC<AccessibleIconOwnProps> = ({ children, label }) =
         'aria-hidden': 'true',
         focusable: 'false', // See: https://allyjs.io/tutorials/focusing-in-svg.html#making-svg-elements-focusable
       })}
-      <VisuallyHidden>{label}</VisuallyHidden>
+      <VisuallyHiddenPrimitive.Root>{label}</VisuallyHiddenPrimitive.Root>
     </>
   );
 };

--- a/packages/react/accordion/src/Accordion.tsx
+++ b/packages/react/accordion/src/Accordion.tsx
@@ -4,7 +4,7 @@ import { useComposedRefs } from '@radix-ui/react-compose-refs';
 import { composeEventHandlers } from '@radix-ui/primitive';
 import { useControllableState } from '@radix-ui/react-use-controllable-state';
 import { Primitive } from '@radix-ui/react-primitive';
-import { Collapsible, CollapsibleButton, CollapsibleContent } from '@radix-ui/react-collapsible';
+import * as CollapsiblePrimitive from '@radix-ui/react-collapsible';
 import { useId } from '@radix-ui/react-id';
 
 import type * as Polymorphic from '@radix-ui/react-polymorphic';
@@ -257,7 +257,10 @@ const AccordionImpl = React.forwardRef((props, forwardedRef) => {
 const ITEM_NAME = 'AccordionItem';
 
 type AccordionItemOwnProps = Polymorphic.Merge<
-  Omit<Polymorphic.OwnProps<typeof Collapsible>, 'open' | 'defaultOpen' | 'onOpenChange'>,
+  Omit<
+    Polymorphic.OwnProps<typeof CollapsiblePrimitive.Root>,
+    'open' | 'defaultOpen' | 'onOpenChange'
+  >,
   {
     /**
      * Whether or not an accordion item is disabled from user interaction.
@@ -273,7 +276,7 @@ type AccordionItemOwnProps = Polymorphic.Merge<
 >;
 
 type AccordionItemPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof Collapsible>,
+  Polymorphic.IntrinsicElement<typeof CollapsiblePrimitive.Root>,
   AccordionItemOwnProps
 >;
 
@@ -296,7 +299,7 @@ const AccordionItem = React.forwardRef((props, forwardedRef) => {
 
   return (
     <AccordionItemProvider open={open} disabled={disabled} buttonId={buttonId}>
-      <Collapsible
+      <CollapsiblePrimitive.Root
         data-state={open ? 'open' : 'closed'}
         disabled={disabled}
         {...accordionItemProps}
@@ -355,9 +358,9 @@ AccordionHeader.displayName = HEADER_NAME;
 
 const BUTTON_NAME = 'AccordionButton';
 
-type AccordionButtonOwnProps = Polymorphic.OwnProps<typeof CollapsibleButton>;
+type AccordionButtonOwnProps = Polymorphic.OwnProps<typeof CollapsiblePrimitive.Button>;
 type AccordionButtonPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof CollapsibleButton>,
+  Polymorphic.IntrinsicElement<typeof CollapsiblePrimitive.Button>,
   AccordionButtonOwnProps
 >;
 
@@ -369,7 +372,7 @@ const AccordionButton = React.forwardRef((props, forwardedRef) => {
   const { buttonNodesRef } = useAccordionContext(BUTTON_NAME);
   const itemContext = useAccordionItemContext(BUTTON_NAME);
 
-  const ref = React.useRef<React.ElementRef<typeof CollapsibleButton>>(null);
+  const ref = React.useRef<React.ElementRef<typeof CollapsiblePrimitive.Button>>(null);
   const composedRefs = useComposedRefs(ref, forwardedRef);
 
   React.useEffect(() => {
@@ -386,7 +389,7 @@ const AccordionButton = React.forwardRef((props, forwardedRef) => {
   }, [buttonNodesRef]);
 
   return (
-    <CollapsibleButton
+    <CollapsiblePrimitive.Button
       aria-disabled={itemContext.open || undefined}
       id={itemContext.buttonId}
       {...props}
@@ -403,9 +406,9 @@ AccordionButton.displayName = BUTTON_NAME;
 
 const PANEL_NAME = 'AccordionPanel';
 
-type AccordionPanelOwnProps = Polymorphic.OwnProps<typeof CollapsibleContent>;
+type AccordionPanelOwnProps = Polymorphic.OwnProps<typeof CollapsiblePrimitive.Content>;
 type AccordionPanelPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof CollapsibleContent>,
+  Polymorphic.IntrinsicElement<typeof CollapsiblePrimitive.Content>,
   AccordionPanelOwnProps
 >;
 /**
@@ -414,7 +417,7 @@ type AccordionPanelPrimitive = Polymorphic.ForwardRefComponent<
 const AccordionPanel = React.forwardRef((props, forwardedRef) => {
   const itemContext = useAccordionItemContext(PANEL_NAME);
   return (
-    <CollapsibleContent
+    <CollapsiblePrimitive.Content
       role="region"
       aria-labelledby={itemContext.buttonId}
       {...props}

--- a/packages/react/alert-dialog/src/AlertDialog.tsx
+++ b/packages/react/alert-dialog/src/AlertDialog.tsx
@@ -3,13 +3,7 @@ import { createContext } from '@radix-ui/react-context';
 import { useComposedRefs } from '@radix-ui/react-compose-refs';
 import { composeEventHandlers } from '@radix-ui/primitive';
 import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
-import {
-  Dialog,
-  DialogOverlay,
-  DialogContent,
-  DialogTrigger,
-  DialogClose,
-} from '@radix-ui/react-dialog';
+import * as DialogPrimitive from '@radix-ui/react-dialog';
 import { useId } from '@radix-ui/react-id';
 import { Slottable } from '@radix-ui/react-slot';
 
@@ -30,9 +24,9 @@ const [AlertDialogProvider, useAlertDialogContext] = createContext<AlertDialogCo
   ROOT_NAME
 );
 
-const AlertDialog: React.FC<React.ComponentProps<typeof Dialog>> = (props) => (
+const AlertDialog: React.FC<React.ComponentProps<typeof DialogPrimitive.Root>> = (props) => (
   <AlertDialogProvider descriptionId={useId()} titleId={useId()}>
-    <Dialog {...props} />
+    <DialogPrimitive.Root {...props} />
   </AlertDialogProvider>
 );
 
@@ -54,12 +48,12 @@ const [
 ] = createContext<AlertDialogContentContextValue>(CONTENT_NAME);
 
 type AlertDialogContentOwnProps = Omit<
-  Polymorphic.OwnProps<typeof DialogContent>,
+  Polymorphic.OwnProps<typeof DialogPrimitive.Content>,
   'refToFocusOnOpen' | 'id'
 >;
 
 type AlertDialogContentPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof DialogContent>,
+  Polymorphic.IntrinsicElement<typeof DialogPrimitive.Content>,
   AlertDialogContentOwnProps
 >;
 
@@ -75,7 +69,7 @@ const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
   const cancelRef = React.useRef<React.ElementRef<typeof AlertDialogCancel> | null>(null);
 
   return (
-    <DialogContent
+    <DialogPrimitive.Content
       role="alertdialog"
       aria-describedby={ariaDescribedBy || descriptionId}
       // If `aria-label` is set, ensure `aria-labelledby` is undefined as to avoid confusion.
@@ -100,7 +94,7 @@ const AlertDialogContent = React.forwardRef((props, forwardedRef) => {
          */}
         <Slottable>{children}</Slottable>
       </AlertDialogContentProvider>
-    </DialogContent>
+    </DialogPrimitive.Content>
   );
 }) as AlertDialogContentPrimitive;
 
@@ -159,25 +153,25 @@ AlertDialogDescription.displayName = DESCRIPTION_NAME;
 
 const CANCEL_NAME = 'AlertDialogCancel';
 
-type AlertDialogCancelOwnProps = Polymorphic.OwnProps<typeof DialogClose>;
+type AlertDialogCancelOwnProps = Polymorphic.OwnProps<typeof DialogPrimitive.Close>;
 type AlertDialogCancelPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof DialogClose>,
+  Polymorphic.IntrinsicElement<typeof DialogPrimitive.Close>,
   AlertDialogCancelOwnProps
 >;
 
 const AlertDialogCancel = React.forwardRef((props, forwardedRef) => {
   const { cancelRef } = useAlertDialogContentContext(CANCEL_NAME);
   const ref = useComposedRefs(forwardedRef, cancelRef);
-  return <DialogClose {...props} ref={ref} />;
+  return <DialogPrimitive.Close {...props} ref={ref} />;
 }) as AlertDialogCancelPrimitive;
 
 AlertDialogCancel.displayName = CANCEL_NAME;
 
 /* ---------------------------------------------------------------------------------------------- */
 
-const AlertDialogTrigger = extendPrimitive(DialogTrigger, 'AlertDialogTrigger');
-const AlertDialogOverlay = extendPrimitive(DialogOverlay, 'AlertDialogOverlay');
-const AlertDialogAction = extendPrimitive(DialogClose, 'AlertDialogAction');
+const AlertDialogTrigger = extendPrimitive(DialogPrimitive.Trigger, 'AlertDialogTrigger');
+const AlertDialogOverlay = extendPrimitive(DialogPrimitive.Overlay, 'AlertDialogOverlay');
+const AlertDialogAction = extendPrimitive(DialogPrimitive.Close, 'AlertDialogAction');
 
 /* ---------------------------------------------------------------------------------------------- */
 

--- a/packages/react/popper/src/Popper.tsx
+++ b/packages/react/popper/src/Popper.tsx
@@ -5,7 +5,7 @@ import { createContext } from '@radix-ui/react-context';
 import { useRect } from '@radix-ui/react-use-rect';
 import { useSize } from '@radix-ui/react-use-size';
 import { Primitive } from '@radix-ui/react-primitive';
-import { Arrow as ArrowPrimitive } from '@radix-ui/react-arrow';
+import * as ArrowPrimitive from '@radix-ui/react-arrow';
 
 import type * as Polymorphic from '@radix-ui/react-polymorphic';
 import type { Side, Align } from '@radix-ui/popper';
@@ -119,12 +119,12 @@ Popper.displayName = POPPER_NAME;
 const ARROW_NAME = 'PopperArrow';
 
 type PopperArrowOwnProps = Polymorphic.Merge<
-  Polymorphic.OwnProps<typeof ArrowPrimitive>,
+  Polymorphic.OwnProps<typeof ArrowPrimitive.Root>,
   { offset?: number }
 >;
 
 type PopperArrowPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof ArrowPrimitive>,
+  Polymorphic.IntrinsicElement<typeof ArrowPrimitive.Root>,
   PopperArrowOwnProps
 >;
 
@@ -149,7 +149,7 @@ const PopperArrow = React.forwardRef(function PopperArrow(props, forwardedRef) {
           pointerEvents: 'auto',
         }}
       >
-        <ArrowPrimitive
+        <ArrowPrimitive.Root
           {...arrowProps}
           ref={forwardedRef}
           style={{

--- a/packages/react/toolbar/src/Toolbar.tsx
+++ b/packages/react/toolbar/src/Toolbar.tsx
@@ -4,7 +4,7 @@ import { createContext } from '@radix-ui/react-context';
 import { RovingFocusGroup, useRovingFocus } from '@radix-ui/react-roving-focus';
 import { Primitive } from '@radix-ui/react-primitive';
 import { Slot } from '@radix-ui/react-slot';
-import { Separator as SeparatorPrimitive } from '@radix-ui/react-separator';
+import * as SeparatorPrimitive from '@radix-ui/react-separator';
 import * as RadioGroupPrimitive from '@radix-ui/react-radio-group';
 
 import type * as Polymorphic from '@radix-ui/react-polymorphic';
@@ -75,9 +75,9 @@ const Toolbar = React.forwardRef((props, forwardedRef) => {
 
 const SEPARATOR_NAME = 'ToolbarSeparator';
 
-type ToolbarSeparatorOwnProps = Polymorphic.OwnProps<typeof SeparatorPrimitive>;
+type ToolbarSeparatorOwnProps = Polymorphic.OwnProps<typeof SeparatorPrimitive.Root>;
 type ToolbarSeparatorPrimitive = Polymorphic.ForwardRefComponent<
-  Polymorphic.IntrinsicElement<typeof SeparatorPrimitive>,
+  Polymorphic.IntrinsicElement<typeof SeparatorPrimitive.Root>,
   ToolbarSeparatorOwnProps
 >;
 
@@ -85,7 +85,7 @@ const ToolbarSeparator = React.forwardRef((props, forwardedRef) => {
   const context = useToolbarContext(SEPARATOR_NAME);
 
   return (
-    <SeparatorPrimitive
+    <SeparatorPrimitive.Root
       orientation={context.orientation === 'horizontal' ? 'vertical' : 'horizontal'}
       {...props}
       ref={forwardedRef}

--- a/packages/react/tooltip/src/Tooltip.tsx
+++ b/packages/react/tooltip/src/Tooltip.tsx
@@ -10,7 +10,7 @@ import { Primitive, extendPrimitive } from '@radix-ui/react-primitive';
 import * as PopperPrimitive from '@radix-ui/react-popper';
 import { Portal } from '@radix-ui/react-portal';
 import { Slottable } from '@radix-ui/react-slot';
-import { VisuallyHidden } from '@radix-ui/react-visually-hidden';
+import * as VisuallyHiddenPrimitive from '@radix-ui/react-visually-hidden';
 import { useId } from '@radix-ui/react-id';
 import { createStateMachine, stateChart } from './machine';
 
@@ -243,9 +243,9 @@ const TooltipContentImpl = React.forwardRef((props, forwardedRef) => {
         }}
       >
         <Slottable>{children}</Slottable>
-        <VisuallyHidden id={context.contentId} role="tooltip">
+        <VisuallyHiddenPrimitive.Root id={context.contentId} role="tooltip">
           {ariaLabel || children}
-        </VisuallyHidden>
+        </VisuallyHiddenPrimitive.Root>
       </PopperPrimitive.Root>
     </PortalWrapper>
   );

